### ISSUE TITLE
feature: track playing speed

### ISSUE THREAD

**Issue by jessicajonesunitedkingdom** [emoon/HippoPlayer#112]
hi

Not sure if this is possible but I sometimes like it to play a tracker music file a bit faster. Currently I have no player that can do this so I export to WAV and play them with a DJ application which can adjust tempo.

Do you think it's possible to make a sort of slider that can be set to 100% (normal speed), or 50% (slower), or 150%, 200%?
Disabled for files/libraries who don't work with this.

**Comment by emoon** [comment]
Yes this is possible to do on some players. It's possible to solve it for other players as well that doesn't support it by "generating" the full song (or to a certain time like 5 min) in memory and when the generation is done then one can switch to faster speed.

This is a bit more complicated to implement as you want playback to start as soon as possible, but then having the player generating more data and then when the full generation is done faster playback can be done.

**Comment by jessicajonesunitedkingdom** [comment]
Maybe there is a library that can speed up/down WAV playback?
The player can for example export the whole song to WAV (normally very fast) and just adjusts speed of this WAV file.
When another song is played, the temporary WAV file is just overwritten.

If the setting is disabled, the player plays normally.

**Comment by emoon** [comment]
That is what I mean generating the full song (meaning Hippo will tell the player to generate the full song instead of bit by bit which will consume more CPU time for a while.  Also depends on how much CPU time the player takes in general.

**Comment by emoon** [comment]
There is actually a issue for this already https://github.com/emoon/HippoPlayer/issues/32

**Comment by jessicajonesunitedkingdom** [comment]
Ah sorry, didn't see that.

I just found an interesting library that works on WAV files (open source , cross platform):

https://www.surina.net/soundtouch/

**Comment by jessicajonesunitedkingdom** [comment]
As an additional feature request: this library can calculate BPM. So maybe cool to add a widget that displays BPM.
You can then see it go up/down when adjusting!

**Comment by emoon** [comment]
Working with WAV is really easy so that isn't a problem, but BPM calculation is cool for sure.

**Comment by jessicajonesunitedkingdom** [comment]
Would it be possible to run 2 instances of the hippoplayer?
I tried this for example with other players like audacious and foobar2000 but it's not working.

It could be cool to mix tracker music by opening 2 instances of hippoplayer and adjusting the BPM :-)

**Comment by emoon** [comment]
I just tested and it works fine :)

**Comment by emoon** [comment]
I think the problem comes down to trying to update settings, database and such at the same time from two instances

**Comment by emoon** [comment]
I guess perhaps the second instance can have updates "frozen" or something.
